distribution/src/main/resources/bin/elasticsearch-plugin.bat

```diff
@@ -2,6 +2,13 @@
 
 SETLOCAL enabledelayedexpansion
 
+if NOT DEFINED JAVA_HOME IF EXIST %ProgramData%\Oracle\java\javapath\java.exe (
+    for /f "tokens=2 delims=[]" %%a in ('dir %ProgramData%\Oracle\java\javapath\java.exe') do @set JAVA_EXE=%%a
```

Comment:
Sorry, I don't think this change should be made. `JAVA_EXE` is a non-standard environment variable, and we already have a standard environment variable in `JAVA_HOME` for pointing to Java. We are moving strongly in the direction of having one way to do things, and not multiple ways to do things and I don't see the benefit of multiple ways to point to Java when we already have the fairly standard way of respecting `JAVA_HOME`.